cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest

```diff
@@ -9,7 +9,9 @@
 {% set cloud_config_volume = "" -%}
 
 {% if grains.cloud is defined -%}
-  {% set cloud_provider = "--cloud-provider=" + grains.cloud -%}
+  {% if grains.cloud != 'vagrant' -%}
```

Comment:
The only feature that vagrant cloud provider had was listing instances using salt master, which is incompativle with masterless salt.

https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/vagrant/vagrant.go#L37

I'm still trying to figure out what to do with the vagrant cloudprovider or if it's necessary.